_posts/2021-11-21-dd_00003_logic_gates_1.md

```diff
@@ -0,0 +1,357 @@
+Z will only be "true" when both A _and_ B are "true"; hence "and".
+
+The behavior of any logical operation can be represented using a special table,
+called the **truth table**. The truth table shows for all possible input
+combinations the corresponding output values, thereby fully characterizing the
+function of the particular logic operation. For example, the truth table of the
+AND operation looks like this:
+
+| A   | B   | Z = A _and_ B |
+| --- | --- | :-----------: |
+| 0   | 0   |       0       |
+| 0   | 1   |       0       |
+| 1   | 0   |       0       |
+| 1   | 1   |       1       |
+
+As you can see in the last row, the output Z is only "on" (1) when _both_
+inputs, A and B, are "on"; otherwise, the output stays "off" (0). Let's look at
+the truth table of another logic operation, the **OR**:
+
+| A   | B   | Z = A _or_ B |
+| --- | --- | :----------: |
+| 0   | 0   |      0       |
+| 0   | 1   |      1       |
+| 1   | 0   |      1       |
+| 1   | 1   |      1       |
+
+Can you see the difference of the OR compared to the AND operation? The OR
+operation produces a "1" at its output when _at least_ one input is set to "1".
+This behavior is also what its name suggests: A _or_ B (and that also includes
+the case when both A and B are 1). Another interesting logic operation is the
+**NOT** operation:
+
+| A   | Z = _not_ A |
+| --- | :---------: |
+| 0   |      1      |
+| 1   |      0      |
+
+The NOT operation only has one input and one output. Again, as the name
+suggests, the NOT operation simply produces the _inverse_ (or the _negation_) of
+its input.
+
+Now, buckle up! Your mind is about to be blown away: With only the three logic
+operations we presented you above, namely the AND, OR, and the NOT operations,
+it is possible to construct ANY digital circuit, even the processor you are
+using to read this blog post[^1]!
+
+Next, we will learn how this is done in practice.
+
+[^1]:
+    For the people that are already screaming: Yes, you need a bit more than
+    just logic operations to build a processor, but for the digital part of the
+    chip, the statement is true.
+
+## Basic Logic Gates
+
+Logic operations are implemented in hardware using **logic gates**. Just like a
+logic operation, a logic gate has inputs and output(s). For example, this is the
+logic gate for the AND operation, or short, **AND gate**:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-and-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+To be fair, this is not how the AND gate looks on the final chip, in reality,
+but this is how we represent an AND gate when drawing logic circuit
+diagrams[^2]. As we mentioned initially, we will see in one of the next post how
+logic gates are really implemented in hardware. For now, it is sufficient to
+view logic gates as _black boxes_ that take inputs, perform logical operations,
```

Comment:
```suggestion
view logic gates as **black boxes** that take inputs, perform logical operations,
```

Reply:
should it be bold as it is a very important and common concept?

Reply:
ok